lib/rom/command.rb

```diff
@@ -84,7 +86,25 @@ def execute(*)
     #
     # @api public
     def call(*args, &block)
-      tuples = execute(*(curry_args + args), &block)
+      tuples =
+        if hooks?
+          prepared =
+            if curried?
+              apply_hooks(before_hooks, *curry_args, *args)
+            else
+              apply_hooks(before_hooks, *args)
+            end
+
+          result = prepared ? execute(prepared, &block) : execute(&block)
+
+          if curried?
+            apply_hooks(after_hooks, result, *args)
+          else
+            apply_hooks(after_hooks, result)
+          end
+        else
+          execute(*(curry_args + args), &block)
```

Comment:
so you do
```ruby
apply_hooks(before_hooks, *curry_args, *args)
```
and then
```ruby
execute(*(curry_args + args), &block)
```
I was curious which one is faster and found that `*(a + b)` is faster than `*a, *b` by ~20 percent in all the cases. :chart_with_upwards_trend: